kv/kvnemesis: TestKVNemesisMultiNode failed

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3831142&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3831142&tab=artifacts#/) on master @ [2b8b0f64c63a376784926667f5fbffa4c4fcc60b](https://github.com/cockroachdb/cockroach/commits/2b8b0f64c63a376784926667f5fbffa4c4fcc60b):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Run
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:679
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyBatchOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:272
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyClientOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:228
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:121
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:961
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:112
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*Applier).Apply
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:66
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis.func2
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:63
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) secondary error attachment
        	            	  | interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | (1) interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | Error types: (1) *runtime.TypeAssertionError
        	            	Wraps: (3) panic applying step 
        	            	  | db3.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
        	            	  |   txn.Get(ctx, /Table/50/"2a719a1c") // (nil, nil)
        	            	  |   txn.ScanForUpdate(ctx, /Table/50/"46d9bbf5", /Table/50/"be8453f8", 0) // (["�4d343bcf ":"v-12", "�52f3ce73 ":"v-14"], nil)
        	            	  |   {
        	            	  |     b := &Batch{}
        	            	  |     b.GetForUpdate(ctx, /Table/50/"1d89bdad")
        	            	  |     b.ReverseScanForUpdate(ctx, /Table/50/"40d3be3b", /Table/50/"42e09018")
        	            	  |     txn.Run(ctx, b)
        	            	  |   }
        	            	  |   return errors.New("rollback")
        	            	  | }): interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode684931478
--- FAIL: TestKVNemesisMultiNode (1.64s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3838086&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3838086&tab=artifacts#/) on master @ [d6e53780641e0c6bb632515936b68378be600b6b](https://github.com/cockroachdb/cockroach/commits/d6e53780641e0c6bb632515936b68378be600b6b):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyBatchOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:272
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:129
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:961
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:112
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*Applier).Apply
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:66
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis.func2
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:63
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) secondary error attachment
        	            	  | interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | (1) interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | Error types: (1) *runtime.TypeAssertionError
        	            	Wraps: (3) panic applying step 
        	            	  | db1.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
        	            	  |   txn.Del(ctx, /Table/50/"e33172bc") // nil
        	            	  |   {
        	            	  |     b := &Batch{}
        	            	  |     b.ReverseScan(ctx, /Table/50/"86da646e", /Table/50/"a2fb9799") // ([], nil)
        	            	  |     b.Put(ctx, /Table/50/"657ae823", v-3) // nil
        	            	  |     txn.Run(ctx, b) // nil
        	            	  |   }
        	            	  |   txn.Del(ctx, /Table/50/"b9af2442") // nil
        	            	  |   b := &Batch{}
        	            	  |   b.ReverseScan(ctx, /Table/50/"9d3f7618", /Table/50/"e10579a2")
        	            	  |   b.GetForUpdate(ctx, /Table/50/"2cdbfefc")
        	            	  |   txn.CommitInBatch(ctx, b)
        	            	  |   return nil
        	            	  | }): interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode11935887
--- FAIL: TestKVNemesisMultiNode (1.50s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-json
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3845254&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3845254&tab=artifacts#/) on master @ [506d129f5f187134c35e2f71860490e044fde989](https://github.com/cockroachdb/cockroach/commits/506d129f5f187134c35e2f71860490e044fde989):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:779
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Run
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:679
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).CommitInBatch
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:745
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyBatchOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:272
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:129
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:961
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:112
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*Applier).Apply
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:66
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis.func2
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:63
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) secondary error attachment
        	            	  | interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | (1) interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | Error types: (1) *runtime.TypeAssertionError
        	            	Wraps: (3) panic applying step 
        	            	  | db1.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
        	            	  |   txn.Put(ctx, /Table/50/"57f90410", v-3) // nil
        	            	  |   txn.Del(ctx, /Table/50/"f2bbcc09") // nil
        	            	  |   b := &Batch{}
        	            	  |   b.ReverseScanForUpdate(ctx, /Table/50/"d397d218", /Table/50/"d7bace4b")
        	            	  |   b.GetForUpdate(ctx, /Table/50/"5d1b2a5d")
        	            	  |   txn.CommitInBatch(ctx, b)
        	            	  |   return nil
        	            	  | }): interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode1794990071
--- FAIL: TestKVNemesisMultiNode (1.96s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3848525&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3848525&tab=artifacts#/) on master @ [8b696d17241b2eb2d333e23750cf6fdc433897d8](https://github.com/cockroachdb/cockroach/commits/8b696d17241b2eb2d333e23750cf6fdc433897d8):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyBatchOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:272
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:129
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:961
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:112
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*Applier).Apply
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:66
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis.func2
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:63
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) secondary error attachment
        	            	  | interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | (1) interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | Error types: (1) *runtime.TypeAssertionError
        	            	Wraps: (3) panic applying step 
        	            	  | db3.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
        	            	  |   {
        	            	  |     b := &Batch{}
        	            	  |     b.Put(ctx, /Table/50/"f3119db8", v-2) // nil
        	            	  |     b.Scan(ctx, /Table/50/"a1bbe0ba", /Table/50/"d052953c") // ([], nil)
        	            	  |     b.Put(ctx, /Table/50/"57dd5a1b", v-3) // nil
        	            	  |     txn.Run(ctx, b) // nil
        	            	  |   }
        	            	  |   b := &Batch{}
        	            	  |   b.Del(ctx, /Table/50/"9df2b40a")
        	            	  |   b.ReverseScanForUpdate(ctx, /Table/50/"359f0248", /Table/50/"e8b94272")
        	            	  |   b.GetForUpdate(ctx, /Table/50/"07cf2228")
        	            	  |   txn.CommitInBatch(ctx, b)
        	            	  |   return nil
        	            	  | }): interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode2344136698
--- FAIL: TestKVNemesisMultiNode (1.48s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- TAGS=deadlock

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3848524&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3848524&tab=artifacts#/) on master @ [8b696d17241b2eb2d333e23750cf6fdc433897d8](https://github.com/cockroachdb/cockroach/commits/8b696d17241b2eb2d333e23750cf6fdc433897d8):


```
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/batch.go:193
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.sendAndFill
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:779
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).Run
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:679
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).CommitInBatch
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:745
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyBatchOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:272
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:129
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:961
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:112
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*Applier).Apply
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:66
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis.func2
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:63
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) secondary error attachment
        	            	  | interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | (1) interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | Error types: (1) *runtime.TypeAssertionError
        	            	Wraps: (3) panic applying step 
        	            	  | db0.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
        	            	  |   b := &Batch{}
        	            	  |   b.ReverseScanForUpdate(ctx, /Table/50/"840090c0", /Table/50/"f40383ad")
        	            	  |   b.GetForUpdate(ctx, /Table/50/"0265883e")
        	            	  |   b.Put(ctx, /Table/50/"04db4012", v-7)
        	            	  |   txn.CommitInBatch(ctx, b)
        	            	  |   return nil
        	            	  | }): interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode690693459
--- FAIL: TestKVNemesisMultiNode (1.16s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

kv/kvnemesis.TestKVNemesisMultiNode [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3852659&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3852659&tab=artifacts#/) on master @ [04eccce27670e9db0a0ba7d9c80aec7ea3e02445](https://github.com/cockroachdb/cockroach/commits/04eccce27670e9db0a0ba7d9c80aec7ea3e02445):


```
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyBatchOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:272
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:129
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:849
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*Txn).exec
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/txn.go:961
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.runTxn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:848
        	            	  | github.com/cockroachdb/cockroach/pkg/kv.(*DB).Txn
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/db.go:830
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.applyOp
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:112
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.(*Applier).Apply
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/applier.go:66
        	            	  | github.com/cockroachdb/cockroach/pkg/kv/kvnemesis.RunNemesis.func2
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/kv/kvnemesis/kvnemesis.go:63
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.GroupWorkers.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:175
        	            	  | github.com/cockroachdb/cockroach/pkg/util/ctxgroup.Group.GoCtx.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/pkg/util/ctxgroup/ctxgroup.go:166
        	            	  | golang.org/x/sync/errgroup.(*Group).Go.func1
        	            	  | 	/go/src/github.com/cockroachdb/cockroach/vendor/golang.org/x/sync/errgroup/errgroup.go:57
        	            	  | runtime.goexit
        	            	  | 	/usr/local/go/src/runtime/asm_amd64.s:1581
        	            	Wraps: (2) secondary error attachment
        	            	  | interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | (1) interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	  | Error types: (1) *runtime.TypeAssertionError
        	            	Wraps: (3) panic applying step 
        	            	  | db3.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
        	            	  |   txn.Del(ctx, /Table/50/"38284117") // nil
        	            	  |   {
        	            	  |     b := &Batch{}
        	            	  |     b.ReverseScan(ctx, /Table/50/"b1063f03", /Table/50/"d1c7c83f") // ([], nil)
        	            	  |     b.GetForUpdate(ctx, /Table/50/"ae532a17") // (nil, nil)
        	            	  |     txn.Run(ctx, b) // nil
        	            	  |   }
        	            	  |   txn.ScanForUpdate(ctx, /Table/50/"1e6bf1b3", /Table/50/"680075db", 0) // (["�25263931 ":"v-1", "�615a3f05 ":"v-3"], nil)
        	            	  |   b := &Batch{}
        	            	  |   b.GetForUpdate(ctx, /Table/50/"25263931")
        	            	  |   b.ReverseScanForUpdate(ctx, /Table/50/"70f8b4be", /Table/50/"fa3b2950")
        	            	  |   txn.CommitInBatch(ctx, b)
        	            	  |   return nil
        	            	  | }): interface conversion: roachpb.Response is nil, not *roachpb.GetResponse
        	            	Error types: (1) *withstack.withStack (2) *secondary.withSecondaryError (3) *errutil.leafError
    panic.go:642: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestKVNemesisMultiNode4190645796
--- FAIL: TestKVNemesisMultiNode (1.36s)
```
<details><summary>Help</summary>
<p>
See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)Parameters in this failure:

- GOFLAGS=-parallel=4
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestKVNemesisMultiNode.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvclient/kvcoord/batch.go
+++ b/pkg/kv/kvclient/kvcoord/batch.go
@@ -144,18 +144,19 @@ func prev(reqs []roachpb.RequestUnion, k roachpb.RKey) (roachpb.RKey, error) {
 		}
 		endKey := h.EndKey
 		if len(endKey) == 0 {
-			// This is unintuitive, but if we have a point request at `x=k` then that request has
-			// already been satisfied (since the batch has already been executed for all keys `>=
-			// k`). We treat `k` as `[k,k)` which does the right thing below. It also does when `x >
-			// k` and `x < k`, so we're good.
+			// If we have a point request for `x < k` then that request has not been
+			// satisfied (since the batch has only been executed for keys `>=k`). We
+			// treat `x` as `[x, x.Next())` which does the right thing below. This
+			// also works when `x > k` or `x=k` as the logic below will skip `x`.
 			//
-			// Note that if `x` is /Local/k/something, then AddrUpperBound below will turn it into
-			// `k\x00`, and so we're looking at the key range `[k, k\x00)`. This is exactly what we
-			// want since otherwise the result would be `k` and so the caller would restrict itself
-			// to `key < k`, but that excludes `k` itself and thus all local keys attached to it.
+			// Note that if the key is /Local/x/something, then instead of using
+			// /Local/x/something.Next() as the end key, we rely on AddrUpperBound to
+			// handle local keys. In particular, AddrUpperBound will turn it into
+			// `x\x00`, so we're looking at the key-range `[x, x.Next())`. This is
+			// exactly what we want as the local key is contained in that range.
 			//
-			// See TestBatchPrevNext for a test case with commentary.
-			endKey = h.Key
+			// See TestBatchPrevNext for test cases with commentary.
+			endKey = h.Key.Next()
 		}
 		eAddr, err := keys.AddrUpperBound(endKey)
 		if err != nil {

--- a/pkg/kv/kvclient/kvcoord/batch_test.go
+++ b/pkg/kv/kvclient/kvcoord/batch_test.go
@@ -49,6 +49,14 @@ func TestBatchPrevNext(t *testing.T) {
 			expFW: max,
 			expBW: min,
 		},
+		{
+			spans: span("a"), key: "a",
+			// Done with `key < a`, so `a <= key` is next.
+			expFW: "a",
+			// Done with `key >= a`, and there's nothing in `[min, a]`, so we return
+			// min here.
+			expBW: min,
+		},
 		{spans: span("a", "b", "c", "d"), key: "c",
 			// Done with `key < c`, so `c <= key` next.
 			expFW: "c",
@@ -87,19 +95,24 @@ func TestBatchPrevNext(t *testing.T) {
 		{spans: abc, key: "b",
 			// We've seen `key < b` so we still need to hit `b`.
 			expFW: "b",
-			// We've seen `key >= b`, so hop over the gap to `a`. Similar to the
-			// first test case.
-			expBW: "a",
+			// We've seen `key >=b`, so hop over the gap to `a`, similar to the first
+			// test case. [`a`] is represented as [`a`,`a.Next()`), so we expect prev
+			// to return a.Next() here..
+			expBW: "a\x00",
 		},
 		{spans: abc, key: "b\x00",
 			// No surprises.
 			expFW: "c",
-			expBW: "b",
+			// Similar to the explanation above, we expect [`b`] = [`b`, `b.Next()`)
+			// here.
+			expBW: "b\x00",
 		},
 		{spans: abc, key: "bb",
 			// Ditto.
 			expFW: "c",
-			expBW: "b",
+			// Similar to the explanation above, we expect [`b`] = [`b`, `b.Next()`)
+			// here.
+			expBW: "b\x00",
 		},
 
 		// Multiple candidates. No surprises, just a sanity check.
@@ -148,6 +161,17 @@ func TestBatchPrevNext(t *testing.T) {
 			// how this doesn't return `b` which would be incorrect as `[KeyMin, b)` does not
 			// contain `loc(b)`.
 			expBW: "b\x00",
+		},
+		{
+			spans: span(loc("a"), "", loc("b"), ""), key: "b",
+			// We've dealt with any key that addresses to `< b`, and `loc(b)` is not
+			// covered by it. `loc(b)` lives between `b` and `b\x00`, so we start at
+			// `b`.
+			expFW: "b",
+			// We've dealt with any key that addresses to `>= b`, which includes
+			// `loc(b)`. The next thing cover is `loc(a)`, which lives between `a`
+			// and `a\x00`, so we return `a\x00` here.
+			expBW: "a\x00",
 		},
 
 		// Multiple candidates. No surprises, just a sanity check.

--- a/pkg/kv/kvnemesis/kvnemesis_test.go
+++ b/pkg/kv/kvnemesis/kvnemesis_test.go
@@ -37,7 +37,6 @@ func init() {
 
 func TestKVNemesisSingleNode(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 73373, "flaky test")
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t)
 
@@ -72,7 +71,6 @@ func TestKVNemesisSingleNode(t *testing.T) {
 
 func TestKVNemesisMultiNode(t *testing.T) {
 	defer leaktest.AfterTest(t)()
-	skip.WithIssue(t, 73370, "flaky test")
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t)
 

--- a/pkg/kv/txn_external_test.go
+++ b/pkg/kv/txn_external_test.go
@@ -446,3 +446,65 @@ func testTxnNegotiateAndSendDoesNotBlock(t *testing.T, multiRange, strict, route
 	atomic.StoreInt32(&done, 1)
 	require.NoError(t, g.Wait())
 }
+
+// TestRevScanAndGet tests that Get and ReverseScan requests in the same batch
+// can be executed correctly. See illustration below for the various
+// combinations tested.
+func TestRevScanAndGet(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	ctx := context.Background()
+
+	tci := serverutils.StartNewTestCluster(t, 1, base.TestClusterArgs{})
+	tc := tci.(*testcluster.TestCluster)
+	defer tc.Stopper().Stop(ctx)
+	db := tc.Servers[0].DB()
+
+	require.NoError(t, db.AdminSplit(ctx, "b", hlc.MaxTimestamp))
+	require.NoError(t, db.AdminSplit(ctx, "h", hlc.MaxTimestamp))
+
+	// Setup:
+	// Ranges:      [keyMin-------b) [b--------h) [h------keyMax)
+	// ReverseScan:                     [d-f)
+	// Gets:                     *  *  *  *   *  *   *
+	testCases := []struct {
+		getKey string
+	}{
+		{
+			// Get on a range to the left of the reverse scan.
+			getKey: "a",
+		},
+		{
+			// Get on the left split boundary.
+			getKey: "b",
+		},
+		{
+			// Get on the same range as the reverse scan but to the left.
+			getKey: "c",
+		},
+		{
+			// Get on the same range and enclosed by the reverse scan.
+			getKey: "e",
+		},
+		{
+			// Get on the same range as the reverse scan but to the right.
+			getKey: "g",
+		},
+		{
+			// Get on the right split boundary.
+			getKey: "h",
+		},
+		{
+			// Get on a range to the right of the reverse scan.
+			getKey: "i",
+		},
+	}
+
+	for _, tc := range testCases {
+		txn := db.NewTxn(ctx, "test")
+		b := txn.NewBatch()
+		b.Get(tc.getKey)
+		b.ReverseScan("d", "f")
+		require.NoError(t, txn.Run(ctx, b))
+	}
+}

